server/src/test/java/keywhiz/service/daos/SecretContentDAOMockTest.java

```diff
@@ -0,0 +1,119 @@
+package keywhiz.service.daos;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import keywhiz.KeywhizConfig;
+import keywhiz.jooq.tables.SecretsContent;
+import keywhiz.service.crypto.RowHmacGenerator;
+import org.apache.commons.lang3.StringUtils;
+import org.jooq.DSLContext;
+import org.jooq.Record1;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+import org.jooq.tools.jdbc.MockConnection;
+import org.jooq.tools.jdbc.MockDataProvider;
+import org.jooq.tools.jdbc.MockExecuteContext;
+import org.jooq.tools.jdbc.MockResult;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+
+public class SecretContentDAOMockTest {
+  private static final long NO_SECRET_ID = 0;
+  private static final String NO_HMAC = null;
+  private static final String NO_CREATOR = null;
+  private static final Map<String, String> NO_METADATA = null;
+  private static final long NO_EXPIRY = 0;
+  private static final long NO_NOW = 0;
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void rejectsSecretWithSizeGreaterThanMaximumSecretSize() {
+    Long maximumSecretSizeInBytes = 100L;
+    String secret = StringUtils.repeat("x", (int) (maximumSecretSizeInBytes + 1));
+
+    SecretContentDAO dao = createDaoWithMaximumSecretSize(maximumSecretSizeInBytes);
+
+    thrown.expect(IllegalArgumentException.class);
+
+    createSecret(dao, secret);
+  }
```

Comment:
There are tests for greater than, equal to, less than, and large secrets (65KB) with no limit set. In order to keep the tests readable there's a lot of magic/boilerplate hidden behind helper methods.